app/controllers/posts_controller.rb

```diff
@@ -27,6 +27,16 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    if @post.destroy
+      redirect_to user_posts_path(@post.author_id)
+    else
+      flash.now[:errors] = @post.errors.full_messages
+      render :show
+    end
+  end
+
   private
 
   def post_params
```

Comment:
- [ ] Let's kindly add the l`oad_and_authorize_resource` method in the Post Controller. It will use a before action to load the resource into an instance variable and authorize it for every action. It is done as below :point_down: 

 
```
 class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

Reply:
Hi @ragangithub, @shihabrafio , Please just add the line of code above in the PostsController. After adding the line, push and I'll leave an approve message  here. You're almost done with the project. Everytthing works as expected. After assigning an "admin" role to other users, they were able to manage posts and comments of other users :+1: :rocket: 

Right now this project is approved on your dashboard. It was an error :confused: 

After adding the line of code, continued with the next project :+1: 

Thanks for understanding!

Reply:
HI @FrederickMih 
Thank you for the review you gave us.In your suggestion you told us to add load_and_authorize_resource.I wanted to ask you a question because since this helper method is used for automatically loading and authorizing but why do we need to check for authorizing in the controller because we are supposed to add destroy buthon based on autherization. we need to check it on the view and for that we have implemented can? and the code is working.
Can you please tell us the reason 😄

Reply:
Hi @ragangithub, I think is a best practices to add it. You added destroy button based on authorization and it works. But in case you may add other actions like edit, update, in the future this method is provided to automatically authorize all the other actions in a RESTfull resource controller.

Reply:
Thanks @FrederickMih for the explanation. I have made the necessary changes